lib/ancestry/instance_methods.rb

```diff
@@ -5,7 +5,11 @@ module InstanceMethods
 
     # Validate that the ancestors don't include itself
     def ancestry_exclude_self
-      errors.add(:base, "#{self.class.name.humanize} cannot be a descendant of itself.") if ancestor_ids.include? self.id
+      errors.add(:base, "#{self.class.name.humanize} cannot be a descendant of itself.") if ancestor_ids.include? self.ancestry_id
+    end
+
```

Comment:
if we do this, does this mean that we also need to update the `has_ancestry` option in `user.rb` model to make sure `ancestry_id_column` is pointing to the `id` column or primary key?

or we need to make sure `ancestry_id` will fallback to `id` column if `ancestry_id_column` is not defined?

Reply:
There wasn't any breaking changes with api.  we won't need to change `user.rb`.  we default to `id`